src/Pim/Bundle/ReferenceDataBundle/Elasticsearch/Filter/Attribute/ReferenceDataFilter.php

```diff
@@ -0,0 +1,178 @@
+        $channel = null,
+        $options = []
+    ) {
+        if (null === $this->searchQueryBuilder) {
+            throw new \LogicException('The search query builder is not initialized in the filter.');
+        }
+
+        $this->checkLocaleAndChannel($attribute, $locale, $channel);
+
+        if (Operators::IS_EMPTY !== $operator && Operators::IS_NOT_EMPTY !== $operator) {
+            $this->checkValue($attribute, $values);
+        }
+
+        $attributePath = $this->getAttributePath($attribute, $locale, $channel);
+
+        switch ($operator) {
+            case Operators::IN_LIST:
+                $clause = [
+                    'terms' => [
+                        $attributePath => $values,
+                    ],
+                ];
+                $this->searchQueryBuilder->addFilter($clause);
+                break;
+
+            case Operators::IS_EMPTY:
+                $clause = [
+                    'exists' => [
+                        'field' => $attributePath,
+                    ],
+                ];
+                $this->searchQueryBuilder->addMustNot($clause);
+                break;
+
+            case Operators::IS_NOT_EMPTY:
+                $clause = [
+                    'exists' => [
+                        'field' => $attributePath,
+                    ],
+                ];
+                $this->searchQueryBuilder->addFilter($clause);
+                break;
+
+            case Operators::NOT_IN_LIST:
+                $mustNotClause = [
+                    'terms' => [
+                        $attributePath => $values,
+                    ],
+                ];
+                $filterClause = [
+                    'exists' => [
+                        'field' => $attributePath,
+                    ],
+                ];
+                $this->searchQueryBuilder->addMustNot($mustNotClause);
+                $this->searchQueryBuilder->addFilter($filterClause);
+                break;
+
+            default:
+                throw InvalidOperatorException::notSupported($operator, static::class);
+        }
+
+        return $this;
+    }
+
+    /**
+     * @param AttributeInterface $attribute
+     *
+     * @return bool
+     */
+    public function supportsAttribute(AttributeInterface $attribute)
+    {
+        $referenceDataName = $attribute->getReferenceDataName();
+
+        $isRegistredReferenceData = null !== $referenceDataName &&
+            !empty($referenceDataName) &&
+            null !== $this->registry->get($attribute->getReferenceDataName());
+
+        return $isRegistredReferenceData;
+    }
+
+    /**
+     * Check if values are valid
+     *
+     * @param AttributeInterface $attribute
+     * @param mixed              $values
+     *
+     * @throws ObjectNotFoundException
+     */
+    protected function checkValue(AttributeInterface $attribute, $values)
+    {
+        FieldFilterHelper::checkArray($attribute->getCode(), $values, static::class);
```

Comment:
Idk about suing FieldFilterHelper for attributes also. I personaly use a is_array check.